core/unit_test/TestRangePolicyConstructors.hpp

```diff
@@ -158,6 +166,14 @@ TEST(TEST_CATEGORY_DEATH, range_policy_implicitly_converted_bounds) {
     ASSERT_DEATH({ (void)UIntPolicy(test_val, 10, Kokkos::ChunkSize(2)); },
                  get_error_msg(expected, test_val));
   }
+  {
+    typename IntIndexType::type test_val = -1;
+    ASSERT_NO_THROW((void)IntPolicy(
+        -2,
+        TestIndexConversionCheck<typename IntPolicy::index_type>(test_val)));
+    ASSERT_NO_THROW((void)UIntPolicy(
+        0, TestIndexConversionCheck<typename IntPolicy::index_type>(test_val)));
+  }
```

Comment:
Shouldn't that pass regardless of whether deprecated code 4 is enabled or not?

Reply:
Also I expect we only care about testing that a non-value preserving construction goes through, that is just the 2nd assertion?

Reply:
Fixed the test. Also removed a redundant test.
Whether `KOKKOS_ENABLE_DEPRECATED_CODE_4` is enabled or not doesn't affect the test, but it now will fail to compile without `if constexpr (std::is_convertible_v<member_type, IndexType>)` at the beginning of `check_conversion_safety()`.

Reply:
> Whether `KOKKOS_ENABLE_DEPRECATED_CODE_4` is enabled or not doesn't affect the test, but it now will fail to compile without `if constexpr (std::is_convertible_v<member_type, IndexType>)` at the beginning of `check_conversion_safety()`.

My point is the test you added is guarded by `#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_4` and that it should not be guarded.

Reply:
Moved this out of `#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_4` guard and made into its own test.